scripts/migrations/2024_11_move_obf_opf_opff_products_to_off_dirs.pl

```diff
@@ -0,0 +1,258 @@
+	ensure_dir_created_or_die("/srv/off/products/$dir/$dir2/$dir3");
+	# if there is an existing off directory for this product, move it to deleted-off-products-codes-replaced-by-other-flavors
+	if (-e "/srv/off/products/$dir/$dir2/$dir3/$dir4") {
+		print STDERR "moving existing product on OFF\n";
+		if (
+			dirmove(
+				"/srv/off/products/$dir/$dir2/$dir3/$dir4",
+				"/srv/off/products/deleted-off-products-codes-replaced-by-other-flavors/$dir$dir2$dir3$dir4"
+			)
+			)
+		{
+			print STDERR
+				"moved /srv/off/products/$dir/$dir2/$dir3/$dir4 to /srv/off/products/deleted-off-products-codes-replaced-by-other-flavors/$dir$dir2$dir3$dir4\n";
+		}
+		else {
+			print STDERR
+				"could not move /srv/off/products/$dir/$dir2/$dir3/$dir4 to /srv/off/products/deleted-off-products-codes-replaced-by-other-flavors/$dir$dir2$dir3$dir4: $!\n";
+			die;
+		}
+	}
+	# move the directory to /srv/off/products
+	if (dirmove("/mnt/$flavor/products/$dir/$dir2/$dir3/$dir4", "/srv/off/products/$dir/$dir2/$dir3/$dir4")) {
+		print STDERR "moved /mnt/$flavor/products/$dir/$dir2/$dir3/$dir4 to /srv/off/products/$dir/$dir2/$dir3/$dir4\n";
+	}
+	else {
+		print STDERR
+			"could not move /mnt/$flavor/products/$dir/$dir2/$dir3/$dir4 to /srv/off/products/$dir/$dir2/$dir3/$dir4: $!\n";
+		die;
+	}
+
+	# move images if they exist
+	if (-e "/mnt/$flavor/images/products/$dir/$dir2/$dir3/$dir4") {
+		print STDERR "moving images from /mnt/$flavor/images/products/$dir/$dir2/$dir3/$dir4\n";
+		ensure_dir_created_or_die("/srv/off/html/images/products/$dir/$dir2/$dir3");
+		# if there is an existing off directory for this product, move it to deleted-off-products-codes-replaced-by-other-flavors
+		if (-e "/srv/off/html/images/products/$dir/$dir2/$dir3/$dir4") {
+			print STDERR "moving existing product images on OFF\n";
+			if (
+				dirmove(
+					"/srv/off/html/images/products/$dir/$dir2/$dir3/$dir4",
+					"/srv/off/html/images/products/deleted-off-products-codes-replaced-by-other-flavors/$dir$dir2$dir3$dir4"
+				)
+				)
+			{
+				print STDERR
+					"moved /srv/off/html/images/products/$dir/$dir2/$dir3/$dir4 to /srv/off/html/images/products/deleted-off-products-codes-replaced-by-other-flavors/$dir$dir2$dir3$dir4\n";
+			}
+			else {
+				print STDERR
+					"could not move /srv/off/html/images/products/$dir/$dir2/$dir3/$dir4 to /srv/off/html/images/products/deleted-off-products-codes-replaced-by-other-flavors/$dir$dir2$dir3$dir4: $!\n";
+				die;
+			}
+		}
+		if (
+			dirmove(
+				"/mnt/$flavor/images/products/$dir/$dir2/$dir3/$dir4",
+				"/srv/off/html/images/products/$dir/$dir2/$dir3/$dir4"
+			)
+			)
+		{
+			print STDERR
+				"moved /mnt/$flavor/images/products/$dir/$dir2/$dir3/$dir4 to /srv/off/html/images/products/$dir/$dir2/$dir3/$dir4\n";
+		}
+		else {
+			print STDERR
+				"could not move /mnt/$flavor/images/products/$dir/$dir2/$dir3/$dir4 to /srv/off/html/images/products/$dir/$dir2/$dir3/$dir4: $!\n";
+			die;
+		}
+	}
+	die;
```

Comment:
why do you need this die ?

Reply:
I launched the script product by product for the first products to verify everything was working as expected as I couldn't replicate everything in the dev environment. I removed the die() in opf etc. after that to do all the products.